### ISSUE TITLE
Voting records display votes and at the same time say neurons were ineligible

### ISSUE THREAD

**Issue by JensGroth** [dfinity/nns-dapp#740]
For some proposals, e.g., 55152, the voting record in the NNS dapp shows how my neurons voted, and at the same time say the same neurons were ineligible to vote due to dissolve delay < 6 months.

**Comment by lmuntanerDfinity** [comment]
Can you share a screenshot?

@mstrasinskis is this a known issue?

**Comment by JensGroth** [comment]
Wonder if it is because some data expiring. For the most recent proposals the record is correct, but for older ones it is not. E.g.
<img width="912" alt="Scr
<img width="299" alt="Screen Shot 2022-04-26 at 9 32 53 AM" src="https://user-images.githubusercontent.com/46536510/165246800-eb94a81c-7583-4fe6-a85b-420c438491c3.png">
een Shot 2022-04-26 at 9 33 46 AM" src="https://user-images.githubusercontent.com/46536510/165246791-3e204c58-e3c2-4312-a18e-187797808ef6.png">
The information about neurons not created is correct, but the information about dissolve delays is wrong.

**Comment by JensGroth** [comment]
For 52949 above, there is no voting record displayed. For a more recent rejected proposal I get both voting record and wrong notification of dissolve delay ineligibility. I.e. for 55152 the voting record looks like this:
<img width="56" alt="Screen Shot 2022-04-26 at 9 40 33 AM" src="https://user-images.githubusercontent.com/46536510/165247918-cd746894-162f-436f-b34b-23913db7816d.png">
and the ineligibility message is
<img width="223" alt="Screen Shot 2022-04-26 at 9 40 53 AM" src="https://user-images.githubusercontent.com/46536510/165247996-b9280d09-14c0-4cd0-b9a5-b35f6c383ba0.png">

**Comment by lmuntanerDfinity** [comment]
Could it be that you added the delay after the proposal was created?

In our test environment, we sometimes see this same error. And the cause seems to come from the backend because the ballots are not added for that neuron.

This was the code to check for that: https://github.com/dfinity/nns-js/blob/main/src/utils/neurons.utils.ts#L40

We rely on the ballots to be there if the neuron can vote.

In summary, to help is with it:

* Did you changed the delay after the proposal was created?
* This is mainnet, correct?

**Comment by JensGroth** [comment]
No, the dissolve delays were set to 8 years close to creation and not changed since
Yes, mainnet

**Comment by lmuntanerDfinity** [comment]
We would need to check the response from the backend. When we experienced this problem was because the backend was not returning a voting ballot.

In how many proposals is that happenning?

**Comment by JensGroth** [comment]
Seems to be all old proposals. Both Executed and Rejected (the other variants cannot become too old).

**Comment by lmuntanerDfinity** [comment]
Ok, that's good I guess. Hard to debug, but I'm glad it doesn't prevent you from voting.

**Comment by peterpeterparker** [comment]
This is a display side effect of way the data are provided by the backend governance canister. On a short term we will hide the particular pane if the proposal has settled (PR https://github.com/dfinity/nns-dapp/pull/1638). Information remain visible on the neuron detail page.
